### PR TITLE
Fix cowboy version depending on running otp version

### PR DESCRIPTION
**Pull request by ThomasHaddad**
Signed-off-by: Thomas Shawarma Haddad <shawarma@wttj.co>
Signed-off-by: Stéphane Robino <stephane@wttj.co>

### CODE REVIEW

**Comment by ThomasHaddad** [comment]
@akira Now widely compatible. Took me some time to fix config & specs, i realized that it was broken for a while on travis... now it is back on tracks.
May i suggest to make a major bump in version soon to support only newer elixir versions? Next features will become hard to implement with such a wide range.

**Comment by ThomasHaddad** [comment]
@akira up?

**Comment by akira** [comment]
Sorry, for delayed response - been very busy lately.

Thanks for attempting this fix. Sure version bump comment makes sense at some point. 
I'm just curious, why were all these changes needed? Seems that this redoes a lot of the testing setup, for example the `--no-start` and `sudo` - it wasn't clear why all this was needed for a fix. So that made it harder to review the code - I will comment on some of these in code.

File: .travis.yml [hunk 1]
```diff
@@ -2,16 +2,18 @@ language: elixir
...
+  - 1.6.6
+  - 1.7.3
 otp_release:
   - 17.5
   - 18.0
   - 19.0
 before_script:
   - MIX_ENV=test mix do deps.get
 script:
-  - MIX_ENV=test mix test --no-start
```
Comment (akira) [inline .travis.yml]: why was this needed?

File: .travis.yml [hunk 2]
```diff
@@ -2,16 +2,18 @@ language: elixir
...
+  - MIX_ENV=test mix test
 after_script:
-  - MIX_ENV=test mix coveralls.travis --no-start
+  - MIX_ENV=test mix coveralls.travis
 services:
   - redis-server
-sudo: false
```
Comment (akira) [inline .travis.yml]: Why was this needed?

File: config/test.exs [hunk 3]
```diff
@@ -4,8 +4,8 @@ config :logger, :console,
   format: "\n$date $time [$level]: $message \n"
 
 config :exq,
-  host: String.to_char_list(System.get_env("REDIS_HOST") || "127.0.0.1"),
```
Comment (akira) [inline config/test.exs]: So this no longer supports the env variable? Seems that it would still work with this old code if this wasn't set.

File: .travis.yml (hunk 1 above)
Comment (ThomasHaddad) [reply .travis.yml]: the `--no-start` prevents from starting any extra applications that would be defined... now that with newer elixir versions it is commonly used you can't keep this flag anymore

File: .travis.yml (hunk 2 above)
Comment (ThomasHaddad) [reply .travis.yml]: There is an issue related to this: https://github.com/travis-ci/travis-ci/issues/7941
Redis seems to be broke if you keep `sudo: false`

File: config/test.exs (hunk 3 above)
Comment (ThomasHaddad) [reply config/test.exs]: `String.to_char_list` cannot be used with newer elixir versions, that's why i removed it, plus the fact that no env var were set on travis. If in the future env var are needed, just keep `System.get_env("REDIS_HOST")` as the value will be cast as a string

**Comment by akira** [comment]
👍 Thanks!!
